### PR TITLE
Add with-log

### PR DESCRIPTION
**Pull request by xaviervia**
Ported from https://github.com/xaviervia/react-dream/blob/master/src/internals/doLog.js

### CODE REVIEW

**Comment by deepsweet** [comment]
I'm doing almost the same _right now_ :)

```
git s
A  packages/with-log/demo/index.jsx
A  packages/with-log/package.json
A  packages/with-log/readme.md
A  packages/with-log/src/index.jsx
A  packages/with-log/test/index.jsx
```

**Comment by deepsweet** [comment]
`with-debugger` is on its way as well :)

**Comment by xaviervia** [comment]
LOL, I should have asked 😂 . I thought it was going to take two minutes and then it was too much fun to stop. Great setup in the project, quite a pleasure working with it.

**Comment by deepsweet** [comment]
Thanks for the inspiration in any way :pray: I'll provide handy snippets for Atom as well, `hlog` → `require('@hocs/with-log')()` or something like this. So you can keep it as dev dependency and use as a temp one-liner.

**Comment by deepsweet** [comment]
Just realized that it's much easier to have inline snippets, something like these:

```js
(C) => (p) => { console.log(props); return <C {...p}/>; }
```

```js
(C) => (p) => { debugger; return <C {...p}/>; }
```

So you don't need to import anything (and then remove it after debug is done).

WDYT?

**Comment by xaviervia** [comment]
I think it is still useful. It's true that for IDE shortcuts that add the HOC inline, this suggestion is better than something you have to import; but for the more general use case, I think having a simple way to debugging HOCs without having to remember how to write the HOC that logs is still pretty valuable.

File: packages/with-log/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  "module": "es/index.js",
+  "files": [
+    "dist/",
+    "es/",
+    "lib/"
+  ],
+  "repository": "deepsweet/hocs",
+  "license": {
+    "type": "MIT",
+    "url": "https://github.com/deepsweet/hocs/blob/master/license.md"
```
Comment (deepsweet) [inline packages/with-log/package.json]: This is awesome trick :heart_eyes: Can't find any documentation though.

**Comment by deepsweet** [comment]
https://github.com/deepsweet/hocs/tree/master/packages/with-log

**Pull request closed without merging**